tests/components/side-panel/toolbar.spec.tsx

```diff
@@ -0,0 +1,59 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import "jest-canvas-mock";
+import { Toolbar } from "src/components/side-panel/toolbar";
+import { TestHelper, mockBuilderState } from "src/util/test-helper";
+
+describe("toolbar", () => {
+    afterEach(() => {
+        jest.restoreAllMocks();
+        jest.resetAllMocks();
+    });
+
+    describe("viewToggleMode()", () => {
+        it("should display the relevant mode when toggleView function is called", () => {
+            renderComponent();
+            fireEvent.click(getIconButton());
+            expect(mockToggleMode).toHaveBeenCalled();
+        });
+
+        it("should run the toggle panel hook when the panel is not being shown", () => {
+            renderComponent({
+                builderContext: {
+                    ...mockBuilderState,
+                    showSidePanel: false,
+                },
+            });
+            fireEvent.click(getIconButton());
+            expect(mockTogglePanel).toHaveBeenCalledWith(true);
+            expect(mockToggleMode).toHaveBeenCalled();
+        });
```

Comment:
- description should be `should expand toggle panel if it is collapsed on clicking the mode button`
- there should be another test for `should collapse toggle panel if it is expanded on clicking the mode button`